lib/evaluate.ts

```diff
@@ -116,29 +116,34 @@ type Visitor<T> = (element: T, c: Continuation, cerr: PartialErrorContinuation)
 export const visitArray = <T>(items: T[], fn: Visitor<T>, c: Continuation, cerr: PartialErrorContinuation) => {
   if (items.length === 0) {
     c([]);
-  } else if (items.length === 1) {
-    fn(items[0], (value) => c([value]), cerr);
   } else {
     const schedule = getTrampolineScheduler();
     const visited = new Set();
 
     function loop(index, accumulated: T[]) {
       if (index < items.length) {
-        fn(
-          items[index],
-          function onNextItem(value) {
-            // If true, it means currently may be happening for example a reevaluation of items
-            // from certain index using call/cc. Copy accumulated previously results and ignore their tail
-            // after given index as this reevalution may happen in the middle of an array.
-            if (visited.has(index)) {
-              accumulated = accumulated.slice(0, index);
-            }
-            accumulated.push(value);
-            visited.add(index);
-            schedule(() => loop(index + 1, accumulated));
-          },
-          cerr
-        );
+          //If true, it means an empty item like `[,]`
+          if(items[index] === null){
```

Comment:
Will it work with arrays including `null` for real, like `[null]` etc?

Reply:
yes, beause it is an ast null, not a value null

Reply:
Right... but `visitArray` is not right place to do that, it should be general as it is. It shouldn't check for nulls.

Reply:
as far as i know, ONLY empty array item will cause a null ast node, in fact the null ast node is illogicality, but all ast parser will generate a null ast node when parse an array with empty items

Reply:
`visitArray` is an utility function, it's not created only to support AST nodes (including arrays nodes). Logic for nulls in AST nodes should be handled in ArrayExpression. I think we should map nulls into some special ast node like `NullValue`.
I can take care of it in one of future improvements. I tend to create one huge PR with many improvements, I can include your fix there written in a way I've described.